src/app/archive/archive_lib/processor.ml

```diff
@@ -107,6 +107,44 @@ module Snarked_ledger_hash = struct
           hash
 end
 
+module Epoch_data = struct
+  type t = {seed: string; ledger_hash_id: int}
+
+  let typ =
+    let encode t = Ok (t.seed, t.ledger_hash_id) in
+    let decode (seed, ledger_hash_id) = Ok {seed; ledger_hash_id} in
+    let rep = Caqti_type.(tup2 string int) in
+    Caqti_type.custom ~encode ~decode rep
+
+  let add_if_doesn't_exist (module Conn : CONNECTION)
+      (t : Coda_base.Epoch_data.Value.t) =
+    let open Deferred.Result.Let_syntax in
+    let Coda_base.Epoch_ledger.Poly.{hash; _} =
+      Coda_base.Epoch_data.Poly.ledger t
+    in
+    let%bind ledger_hash_id =
+      Snarked_ledger_hash.add_if_doesn't_exist (module Conn) hash
+    in
+    let seed =
+      Coda_base.Epoch_data.Poly.Stable.V1.seed t
+      |> Snark_params.Tick.Field.to_string
+    in
+    match%bind
+      Conn.find_opt
+        (Caqti_request.find_opt typ Caqti_type.int
+           "SELECT id FROM epoch_ledgers WHERE ledger_hash_id = ? AND seed = ?")
```

Comment:
should this be `epoch_data` instead of `epoch_ledgers` ?

Reply:
Yeah. I forget to rename it here.